torch/autograd/function.py

```diff
@@ -409,6 +363,68 @@ def jvp(ctx: Any, *grad_inputs: Any) -> Any:
         raise NotImplementedError("You must implement the jvp function for custom "
                                   "autograd.Function to use it with forward mode AD.")
 
+
+class Function(_SingleLevelFunction):
+    r"""Base class to create custom `autograd.Function`
+
+    To create a custom `autograd.Function`, subclass this class and implement
+    the :meth:`forward` and :meth:`backward` static methods. Then, to use your custom
+    op in the forward pass, call the class method ``apply``. Do not call
+    :meth:`forward` directly.
+
+    To ensure correctness and best performance, make sure you are calling the
+    correct methods on ``ctx`` and validating your backward function using
+    :func:`torch.autograd.gradcheck`.
+
+    See :ref:`extending-autograd` for more details on how to use this class.
+
+    Examples::
+
+        >>> class Exp(Function):
+        >>>     @staticmethod
+        >>>     def forward(ctx, i):
+        >>>         result = i.exp()
+        >>>         ctx.save_for_backward(result)
+        >>>         return result
+        >>>
+        >>>     @staticmethod
+        >>>     def backward(ctx, grad_output):
+        >>>         result, = ctx.saved_tensors
+        >>>         return grad_output * result
+        >>>
+        >>> # Use it by calling the apply method:
+        >>> # xdoctest: +SKIP
+        >>> output = Exp.apply(input)
+    """
+    def __init__(self, *args, **kwargs):
+        cls = self.__class__
+        warnings.warn(f"{cls} should not be instantiated. Methods on autograd functions"
+                      "are all static, so you should invoke them on the class itself. "
+                      "Instantiating an autograd function will raise an "
+                      "error in a future version of PyTorch.", DeprecationWarning)
+
+    def __call__(self, *args, **kwargs):
+        raise RuntimeError(
+            "Legacy autograd function with non-static forward method is deprecated. "
+            "Please use new-style autograd function with static forward method. "
+            "(Example: https://pytorch.org/docs/stable/autograd.html#torch.autograd.Function)")
+
+    # for the tracer
+    is_traceable = False
+
+    @classmethod
+    def apply(cls, *args, **kwargs):
+        if not torch._C._is_autograd_function_extension_enabled():
+            return super().apply(*args, **kwargs)
+
+        # TODO: fix circular import
+        from torch._functorch.autograd_function import custom_function_call
```

Comment:
I would prefer to handle this in a follow-up:
- due to the feature flag this is not exercised in the fast-path
- It might involve a refactor of torch/_ops.py